packages/headless/src/features/query-suggest/query-suggest-actions.ts

```diff
@@ -166,7 +180,15 @@ export const buildQuerySuggestRequest = async (
     ...(s.configuration.analytics.enabled && {
       visitorId: await getVisitorID(s.configuration.analytics),
       ...(s.configuration.analytics.enabled &&
-        (await fromAnalyticsStateToAnalyticsParams(s.configuration.analytics))),
+      s.configuration.analytics.analyticsMode === 'legacy'
+        ? await legacyFromAnalyticsStateToAnalyticsParams(
+            s.configuration.analytics
+          )
+        : await fromAnalyticsStateToAnalyticsParams(
```

Comment:
Probably doesn't need to be awaited, see my comment here: https://github.com/coveo/ui-kit/pull/3820/files#diff-94c94221a30278160bf3d13828c55f9c85ab3bedb259dbd080231da783a13284L8